File: clients/client-python/gravitino/api/fileset_data_operation.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from enum import Enum


class FilesetDataOperation(Enum):
    """An enum class containing fileset data operations that supported."""

    CREATE = "create"
    OPEN = "open"
    APPEND = "append"
    RENAME = "rename"
    DELETE = "delete"
    GET_FILE_STATUS = "get_file_status"
    LIST_STATUS = "list_status"
    MKDIRS = "mkdirs"
    EXISTS = "exists"
    CREATED_TIME = "created_time"
    MODIFIED_TIME = "modified_time"
    COPY_FILE = "copy_file"
    CAT_FILE = "cat_file"
    GET_FILE = "get_file"
    UNKNOWN = "unknown"

File: api/src/main/java/com/datastrato/gravitino/file/FilesetDataOperation.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.file;

/** An enum class containing fileset data operations that supported. */
public enum FilesetDataOperation {
  CREATE,
  OPEN,
  APPEND,
  RENAME,
  DELETE,
  GET_FILE_STATUS,
  LIST_STATUS,
  MKDIRS,
  TRUNCATE,
  GET_DEFAULT_REPLICATION,
  GET_DEFAULT_BLOCK_SIZE,
  GET_FILE_CHECKSUM,
  SET_WORKING_DIR,
  EXISTS,
  CREATED_TIME,
  MODIFIED_TIME,
  COPY_FILE,
  CAT_FILE,
  GET_FILE,
  UNKNOWN;
}
